Formula/c/chatgpt-cli.rb

```diff
@@ -0,0 +1,32 @@
+class ChatgptCli < Formula
+  desc "CLI for interacting with the OpenAI ChatGPT API"
+  homepage "https://github.com/kardolus/chatgpt-cli"
+  url "https://github.com/kardolus/chatgpt-cli/archive/refs/tags/v1.5.5.tar.gz"
+  sha256 "158620e871411053310c6882a3e164e013389b0b6d99dd6fb42d8d82c988e51d"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    git_commit = "422f3b9dc0b290ed7b7bc86d1d6c83ff79266de0"
+    git_tag = "v1.5.5"
+    ldflags = "-s -w -X main.GitCommit=#{git_commit}"
+    ldflags << " -X main.GitVersion=#{git_tag}"
+
+    cd "cmd/chatgpt" do
+      system "go", "build", "-mod=vendor", "-ldflags=#{ldflags}", "-o", bin/"chatgpt", "."
+    end
+  end
+
+  test do
+    config_output = shell_output("#{bin}/chatgpt --config")
+    config_keys = %w[
+      name api_key model max_tokens context_window role temperature
+      top_p frequency_penalty presence_penalty thread omit_history
+      url completions_path models_path auth_header auth_token_prefix
+    ]
+    config_keys.each do |key|
+      assert_match key, config_output
+    end
+  end
```

Comment:
```suggestion
  def install
    ldflags = %W[
      -s -w
      -X main.GitCommit=#{tap.user}
      -X main.GitVersion=#{version}
    ]
    system "go", "build", *std_go_args(ldflags:, output: bin/"chatgpt"), "./cmd/chatgpt"

    generate_completions_from_executable(bin/"chatgpt", "--set-completions", base_name: "chatgpt")
  end

  test do
    assert_match "name: openai", shell_output("#{bin}/chatgpt --config")
    assert_match version.to_s, shell_output("#{bin}/chatgpt --version")
  end
```

Reply:
I assume that these tests also run when you are upgrading? I could be wrong. But if they do run when you upgrade, then this test can fail for folks who have the CLI configured to run against `azure` or other chatgpt providers. Those users may have updated the name from `openai` to for example `azure`. If `tap.version` revers to the latest git commit hash of my CLI this works. Otherwise, this will in theory be a breaking change.

Reply:
@chenrui333 sorry, forgot to tag you in the reply ^

Reply:
> ChatGPT CLI version v1.5.3 (commit c0a2944abf1d3ac89689f7ee5ca84c7ac0987aed)


we are tracking releases via release tag, not commit, and they are equivalent from release tag perspective.

Reply:
this would allow us to do the source tarball download rather than git checkout

Reply:
@chenrui333 Ah I see. It's just super convenient to have a git hash as an additional identifier. I do so in all my services for example (usually on a health endpoint). And sicne I have been doing so for over a year in my CLI, people may in theory be depending on it. The change you propose changes the behavior of the existing command. I will have to think about it. Don't really want to bump my version number.

Reply:
Also, the test you propose is more brittle than mine. It will break existing installations if it runs on an upgrade.

Reply:
> Also, the test you propose is more brittle than mine. It will break existing installations if it runs on an upgrade.

It only runs on `brew test <formula>`

Reply:
> And sicne I have been doing so for over a year in my CLI, people may in theory be depending on it.

Not in homebrew though. And your current configuration makes it impossible to run `brew bump` on this formula

Reply:
@SMillerDev thanks for the explanation. Does that mean that `brew test` runs in its own environment (with it's own home dir etc). If it does, then the proposed test is fine. If it runs against the users environment and if it runs before an upgrade is done, then it will be a flaky test. Note that I am already extensively testing the CLI on the github side through a github workflow. 

Can you explain the `brew bump` issue? How is `brew` dependent on the output of my `chatgpt --version` command?

Reply:
@SMillerDev fixed the syntax error in the proposed change and adopted the `#{version}` variable. Will this resolve the `brew bump` issue?

Reply:
> Just to clarify, the #{tap.user} part line is the issue. 

oh, it might be the issue as the way that your app wants to consume a actual sha, can we maybe omit entirely? The git sha thing is not needed for go formulae, and we try to avoid using it.